pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -256,7 +215,7 @@ func BuildSidecarOutboundVirtualHosts(node *model.Proxy, push *model.PushContext
 		if listenerPort == 0 {
 			// Take all ports when listen port is 0 (http_proxy or uds)
 			// Expect virtualServices to resolve to right port
-			servicesByName[svc.Hostname] = svc
+			servicesByName[svc.Hostname] = svc.DeepCopy() // To ensure further changes does not affect service registry.
```

Comment:
why?

Reply:
This is not needed for this PR. But in general added it fo consistency with else block. If you think it is not needed, 
I will remove it